editor-core/src/command.rs

```diff
@@ -0,0 +1,399 @@
+    InsertFirstNonBlank,
+    #[strum(serialize = "append")]
+    Append,
+    #[strum(serialize = "append_end_of_line")]
+    AppendEndOfLine,
+    #[strum(serialize = "toggle_visual_mode")]
+    ToggleVisualMode,
+    #[strum(serialize = "toggle_linewise_visual_mode")]
+    ToggleLinewiseVisualMode,
+    #[strum(serialize = "toggle_blockwise_visual_mode")]
+    ToggleBlockwiseVisualMode,
+    #[strum(serialize = "duplicate_line_up")]
+    DuplicateLineUp,
+    #[strum(serialize = "duplicate_line_down")]
+    DuplicateLineDown,
+}
+
+impl EditCommand {
+    pub fn not_changing_buffer(&self) -> bool {
+        matches!(
+            self,
+            &EditCommand::ClipboardCopy
+                | &EditCommand::Yank
+                | &EditCommand::NormalMode
+                | &EditCommand::InsertMode
+                | &EditCommand::InsertFirstNonBlank
+                | &EditCommand::Append
+                | &EditCommand::AppendEndOfLine
+                | &EditCommand::ToggleVisualMode
+                | &EditCommand::ToggleLinewiseVisualMode
+                | &EditCommand::ToggleBlockwiseVisualMode
+        )
+    }
+}
+
+#[derive(
+    Display, EnumString, EnumIter, Clone, PartialEq, Eq, Debug, EnumMessage, IntoStaticStr,
+)]
+pub enum MoveCommand {
+    #[strum(serialize = "down")]
+    Down,
+    #[strum(serialize = "up")]
+    Up,
+    #[strum(serialize = "left")]
+    Left,
+    #[strum(serialize = "right")]
+    Right,
+    #[strum(serialize = "word_backward")]
+    WordBackward,
+    #[strum(serialize = "word_forward")]
+    WordForward,
+    #[strum(serialize = "word_end_forward")]
+    WordEndForward,
+    #[strum(message = "Document Start")]
+    #[strum(serialize = "document_start")]
+    DocumentStart,
+    #[strum(message = "Document End")]
+    #[strum(serialize = "document_end")]
+    DocumentEnd,
+    #[strum(serialize = "line_end")]
+    LineEnd,
+    #[strum(serialize = "line_start")]
+    LineStart,
+    #[strum(serialize = "line_start_non_blank")]
+    LineStartNonBlank,
+    #[strum(serialize = "go_to_line_default_last")]
+    GotoLineDefaultLast,
+    #[strum(serialize = "go_to_line_default_first")]
+    GotoLineDefaultFirst,
+    #[strum(serialize = "match_pairs")]
+    MatchPairs,
+    #[strum(serialize = "next_unmatched_right_bracket")]
+    NextUnmatchedRightBracket,
+    #[strum(serialize = "previous_unmatched_left_bracket")]
+    PreviousUnmatchedLeftBracket,
+    #[strum(serialize = "next_unmatched_right_curly_bracket")]
+    NextUnmatchedRightCurlyBracket,
+    #[strum(serialize = "previous_unmatched_left_curly_bracket")]
+    PreviousUnmatchedLeftCurlyBracket,
+    #[strum(message = "Paragraph forward")]
+    #[strum(serialize = "paragraph_forward")]
+    ParagraphForward,
+    #[strum(message = "Paragraph backward")]
+    #[strum(serialize = "paragraph_backward")]
+    ParagraphBackward,
+}
+
+impl MoveCommand {
+    pub fn to_movement(&self, count: Option<usize>) -> Movement {
+        use MoveCommand::*;
```

Comment:
Just reading through the code and noticed this. Not necessarily super important. 

Using `use enum::*` can be a pretty bad idea. 

If we were to ever remove one of the variants from MoveCommand it wouldn't cause a compiler error and that variant would then act as a catch all that matches all other patterns. So then if another command was added it would be matched under that variant. 

It's generally better to do `use MoveCommand as MC` or some other short name. This adds the convenience of the short name and doesn't cause the same issues.